db/migrate/20180626125655_rename_user_maintenance_zone_record.rb

```diff
@@ -0,0 +1,31 @@
+class RenameUserMaintenanceZoneRecord < ActiveRecord::Migration[5.0]
+  class Zone < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    validates_presence_of   :name
+
+    MAINTENANCE_ZONE_NAME = "__maintenance__".freeze
+  end
+
+  def up
+    say_with_time("Renaming user-defined Maintenance Zone") do
+      zone = Zone.in_my_region.where(:name => Zone::MAINTENANCE_ZONE_NAME).first
+      if zone.present?
+        zone.name = "#{zone.name}_0"
```

Comment:
Getting into some real edge cases now, but if they had a `__maintenance___0` zone already this will fail.

Reply:
Should we instead generate a unique zone name for the region and store it somewhere in the database (maybe MiqRegion) so that we don't run into naming conflicts?

Reply:
Ok, I'll use "#{Zone::MAINTENANCE_ZONE_NAME}_#{Zone.my_region_number}"
Problem is the same, but risk of fail is extremely low, I think